index.js

```diff
@@ -1,88 +1,119 @@
+console.log("%cQuestion 3", "color: red")
+
+function drinkWater(currentThirstLevel) {
+  console.log ("Man I sure am thirsty")
+  currentThirstLevel = currentThirstLevel - 1
+  console.log("Ahh that hits the spot")
+  return (currentThirstLevel)
+}
+
 
-// console.log(drinkWater(12))
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 
-// ***** Question 4 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red")
 
-// console.log(sameSameButDifferent(5, 5)) 
+function sameSameButDifferent(num, maybeNum) {
+  if (num == parseInt(maybeNum)) {
+    if (typeof(num) == typeof(maybeNum)) {
+      return ("same same")
+    } else {
+      return ("same same (but different)")
+    }
+  }
+  if (num != parseInt(maybeNum)) {
+      return ("different")
+  }
+}
+console.log(sameSameButDifferent(5, 5))
 // // => "same same"
 
-// console.log(sameSameButDifferent(123, "123")) 
+console.log(sameSameButDifferent(123, "123"))
 // // => "same same (but different)"
 
-// console.log(sameSameButDifferent(5, 7)) 
+console.log(sameSameButDifferent(5, 7))
 // // => "different"
 
-// console.log(sameSameButDifferent(123, "122")) 
+console.log(sameSameButDifferent(123, "122"))
 // // => "different"
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red")
 
 
 
-// ***** Question 5 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 5", "color: red")
-// const student1 = { name: "Duane", grade: 88 }
-// updateGrade(student1, 92)
-// console.log(student1)
+
+console.log("%cQuestion 5", "color: red")
+
+function updateGrade(student, grade) {
+  student.grade = grade
+}
+
+
+
+const student1 = { name: "Duane", grade: 88 }
+updateGrade(student1, 92)
+console.log(student1)
 // // => { name: "Duane", grade: 92 }
-// console.log("%c----------", "color: red")
+console.log("%c----------", "color: red")
 
 
 
-// ***** Question 6 *****
-// *** Uncomment the lines below to test
-// console.log("%cQuestion 6", "color: red")
-
-// const users = [ 
-//   { 
-//     name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
-//   },
-//   { 
-//     name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
-//   }
-// ]
-// printNameAndPhones(users)
+
+console.log("%cQuestion 6", "color: red")
+
+function printNameAndPhones(users) {
+  for (let i = 0; i < users.length; i++) {
+    console.log(users[i].name)
+    console.log(`Cell: ${users[i].phones.cell}`)
+    console.log(`Office: ${users[i].phones.office}`)
+  }
+}
+
+const users = [
+  {
+    name: "Duane", phones: { cell: "555-123-4567", office: "555-456-7890" }
+  },
+  {
+    name: "Liza", phones: { cell: "555-234-5678", office: "555-567-1234" }
+  }
+]
+printNameAndPhones(users)
```

Comment:
Wonderful 👏

- Here's how the body of the function for this solution would look with for..of:
```javascript
  for (let user of users) {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  }
```

- Here's how the body of the function for this solution would look with forEach:
```javascript
  users.forEach(function(user) {
    console.log(user.name);
    console.log(`Cell: ${user.phones.cell}`);
    console.log(`Office: ${user.phones.office}`);
  })